libraries/abstractions/common_io/include/iot_spi.h

```diff
@@ -155,86 +163,132 @@ void iot_spi_set_callback( IotSPIHandle_t const pxSPIPeripheral,
+ * - IOT_SPI_INVALID_VALUE, if
+ *     - pxSPIPeripheral is NULL
+ *     - pxSPIPeripheral is not opened yet
+ *     - pucBuffer is NULL with requests which needs buffer
  */
 int32_t iot_spi_ioctl( IotSPIHandle_t const pxSPIPeripheral,
                        IotSPIIoctlRequest_t xSPIRequest,
                        void * const pvBuffer );
 
 /**
- * @brief The SPI master starts reading the number of bytes from the slave
- * synchronously.
+ * @brief The SPI master starts reading from the slave synchronously.
+ *
+ * This function attempts to read certain number of bytes from slave device to a pre-allocated buffer, in synchronous way.
+ * Partial read might happen.
+ * And the number of bytes that have been actually read can be obtained by calling iot_spi_ioctl.
  *
- * The application should call this function to start synchronous read.
- * The application should also give the address of the buffer to read the data
- * into; with number of bytes to read.
+ * @note Dummy data will be written to slave while reading. The dummy data value can be configured with iot_spi_ioctl.
+ *
+ * @warning None of other read or write functions shall be called during this function.
  *
  * @param[in] pxSPIPeripheral The SPI peripheral handle returned in open() call.
  * @param[out] pvBuffer The buffer to store the received data.
  * @param[in] xBytes The number of bytes to read.
  *
- * @return SUCCESS = IOT_SPI_SUCCESS,
- *          FAILED = IOT_SPI_INVALID_VALUE,
- *          SPI_BUSY = IOT_SPI_BUS_BUSY,
- *          Master_UNINITIALIZED = IOT_SPI_MASTER_NOT_INITIALIZED
- *          Slave not selected = IOT_SPI_SLAVE_NOT_SELECTED
+ * @return
+ * - IOT_SPI_SUCCESS, on success (all the requested bytes have been read)
+ * - IOT_SPI_INVALID_VALUE, if
+ *     - pxSPIPeripheral is NULL
+ *     - pxSPIPeripheral is not opened yet
+ *     - pucBuffer is NULL
+ *     - xBytes is 0
+ * - IOT_SPI_READ_ERROR, if there is some unknown driver error.
+ * - IOT_SPI_BUSY, if the bus is busy which means there is an ongoing transaction.
  */
 int32_t iot_spi_read_sync( IotSPIHandle_t const pxSPIPeripheral,
                            uint8_t * const pvBuffer,
                            size_t xBytes );
 
 /**
- * @brief The SPI master starts reading the number of bytes from the slave
- * asynchronously.
+ * @brief The SPI master starts reading from the slave asynchronously.
+ *
+ * This function attempts to read certain number of bytes from a pre-allocated buffer, in asynchronous way.
+ * It returns immediately when the operation is started and the status can be check by calling iot_spi_ioctl.
+ * Once the operation completes, successful or not, the user callback will be invoked.
  *
- * The application should call this function to start synchronous read.
- * The application should also give the address of the buffer to read the data
- * into; with number of bytes to read.
+ * Partial read might happen, e.g. slave device unable to receive more data.
```

Comment:
Similar here. please clarify that Callback will NOT be called on partial reads unless there is a success or error.